gateway/src/apicast/policy/standalone/standalone.lua

```diff
@@ -0,0 +1,364 @@
+  return objects
+end
+
+local Route = { }
+
+local function build_routes(configuration)
+  return build_objects(Route.new, configuration.routes)
+end
+
+local Destination = { }
+do
+  local Destination_mt = { __index = Destination }
+
+  function Destination.new(config)
+    if not config then return nil end
+
+    return setmetatable({
+      service = config.service,
+      http_response = config.http_response,
+    }, Destination_mt)
+  end
+end
+
+local Condition = { }
+do
+
+  local Condition_mt = { __index = Condition }
+
+  local operations = {
+    server_port = function(self) return
+      ngx.var.server_name == self.value or ngx.var.server_port == self.value
+    end,
+    uri_path = function(self) return ngx.var.uri == self.value end,
+    http_method = function(self) return ngx.req.get_method() == self.value end,
+    http_host = function(self) return ngx.var.host == self.value end,
+    always = function(self) return self.value end,
+    unknown = function(self) ngx.log(ngx.ERR, 'unknown condition ', self.name); return end
+  }
+
+  function Condition.new(name, value)
+    return setmetatable({
+        fun = operations[name] or operations.unknown,
+        value = value,
+        name = name,
+    }, Condition_mt)
+  end
+
+  function Condition:match(context)
+    local res = self:fun(context)
+
+    ngx.log(ngx.DEBUG, 'condition ', self.name, ' == ', self.value, ' : ', res)
+
+    return res
+  end
+end
+
+local Match = { }
+do
+  local Match_mt = { __index = Match }
+
+  function Match.new(config)
+    local matchers = { }
+
+    for name, value in pairs(config) do
+      insert(matchers, Condition.new(name, value))
+    end
+
+    return setmetatable(matchers, Match_mt)
+  end
+
+  function Match:any(context)
+    for i=1, #self do
+      if self[i]:match(context) then
+        return self[i]
+      end
+    end
+  end
+
+  function Match:all(context)
+    for i=1, #self do
+      if not self[i]:match(context) then
+        return false
+      end
+    end
+
+    return true
+  end
+end
+
+do
+  local Route_mt = {
+    __index = Route,
+    __tostring = function(route)
+      local match = tab_new(#route.conditions, 0)
+
+      for i=1, #route.conditions do
+        match[i] = format("%s = %s", route.conditions[i].name, route.conditions[i].value)
+      end
+
+      return format('%s', concat(match, ' and '))
+    end
+  }
+
+  function Route.new(config)
+    return setmetatable({
+      name = config.name,
+      conditions = Match.new(config.match),
+      destination = Destination.new(assert(config.destination, 'route is missing destination')),
+      routes = build_routes(config),
+    }, Route_mt)
+  end
+
+  function Route:match(context)
+    return self.conditions:all(context)
+  end
+end
+
+local Service = { }
+
+local function build_services(configuration)
+  return build_objects(Service.new, configuration.internal)
+end
+
+do
+  local Service_mt = { __index = Service }
+
+  local function build_policy_chain(policies)
```

Comment:
It would make sense to have this method in the `PolicyChain` module.

Reply:
Extracted what could be extracted into the `PolicyChain`. But still some had to stay because it depends on the structure of the standalone policy configuration.